activesupport/lib/active_support/ordered_options.rb

```diff
@@ -92,16 +92,38 @@ def inspect
   #   h.boy  # => 'John'
   class InheritableOptions < OrderedOptions
     def initialize(parent = nil)
-      if parent.kind_of?(OrderedOptions)
+      @parent = parent
+      if @parent.kind_of?(OrderedOptions)
         # use the faster _get when dealing with OrderedOptions
-        super() { |h, k| parent._get(k) }
-      elsif parent
-        super() { |h, k| parent[k] }
+        super() { |h, k| @parent._get(k) }
+      elsif @parent
+        super() { |h, k| @parent[k] }
       else
         super()
+        @parent = {}
       end
     end
 
+    def to_h
+      @parent.dup.merge(self)
+    end
+
+    def ==(other)
+      to_h == other.to_h
+    end
+
+    def inspect
+      "#<#{self.class.name} #{to_h.inspect}>"
+    end
+
+    def key?(key)
+      super || @parent.key?(key)
+    end
+
+    def overridden?(key)
+      @parent.key?(key) && keys.include?(key.to_sym)
+    end
```

Comment:
I'm not opposed to the name `overridden?`, but I just want to note that the reason I chose `override?` in #47149 was to match the tense of methods like `Array#include?` and `String#start_with?`.

Reply:
Interesting. I parsed `override?` as "should this override?", so I opted for "is this overridden?". But it's a compelling argument "does this override?", as in "does this include?" and "does this start with?".

I was thinking `overridden?`, like [`frozen?`](https://github.com/rails/rails/blob/6b93fff8af32ef5e91f4ec3cfffb081d0553faf0/activerecord/lib/active_record/core.rb#L599) not (`freeze?`), [`block_given?`](https://docs.ruby-lang.org/en/3.2/Kernel.html#method-i-block_given-3F) (not `block?` or similar), [`attached?`](https://github.com/rails/rails/blob/6b93fff8af32ef5e91f4ec3cfffb081d0553faf0/activestorage/lib/active_storage/attached/many.rb#L66) (not `attach?`), [`cached?`](https://github.com/rails/rails/blob/6b93fff8af32ef5e91f4ec3cfffb081d0553faf0/activerecord/lib/active_record/connection_adapters/schema_cache.rb#L303) (not `cache?`), [`loaded?`](https://api.rubyonrails.org/classes/ActiveRecord/Relation.html#loaded?) (not `load?`), [`destroyed?`](https://github.com/rails/rails/blob/6b93fff8af32ef5e91f4ec3cfffb081d0553faf0/activerecord/lib/active_record/persistence.rb#L682) (not `destroy?`) etc.